File: golang/day1/main.go
package main

import (
	"fmt"
	"io/ioutil"
	"strings"
	"strconv"
)

func report() string {
	data, err := ioutil.ReadFile("./data.txt")

	if err != nil {
		panic(err)
	}

	return string(data)
}

func main(){
	data := strings.Split(report(), "\n")
	higher := make([]int, 0)

	for i := 1; i < len(data); i++ {
		current, _ := strconv.Atoi(data[i])
		previous, _ := strconv.Atoi(data[i-1])

		if(current > previous){
			higher = append(higher, current)
		}
	}

	fmt.Println(len(higher))
}

File: typescript/day1/main.ts
import fs from "fs";

function report(): string {
  let data: string;
  try {
    data = fs.readFileSync('./data.txt', 'utf8')
    return data
  } catch(err) {
    console.log(err)
    return "";
  }
}

const data = report()
  .trim()
  .split("\n")
  .filter((x, i, array) => {
    if(i == 0){
      return false;
    } else if(+x > +array[i-1]){
      return x
    }
  }).length;

console.log(data);
